src/JSInterop/Microsoft.JSInterop/src/DotNetDispatcher.cs

```diff
@@ -354,6 +360,8 @@ private static (MethodInfo, Type[]) GetCachedMethodInfo(AssemblyKey assemblyKey,
             return result;
         }
 
+        internal static bool WorkAroundTestBug;
```

Comment:
I don't understand this. Am I missing something, or is its value always `true`? If so, why do we even have this flag and the `if`/`else` case below? Why not just always do L378?

Reply:
Oh wait, I see how it only gets set to `true` in tests.

Still, is there any disadvantage in always using the logic on L378?

Reply:
Not particularly. I didn't think it was necessary to change product code, but in the most ordinary cases, you're unlikely to see more than two instances of the same assembly so this doesn't change much.